### ISSUE TITLE
Style/RedundantParentheses false positive on a pin operator with expression

### ISSUE THREAD

**Issue by Tietew** [rubocop/rubocop#10979]
Ruby 3.1+ allows expression in pattern match (pin operator).
cf. https://bugs.ruby-lang.org/issues/17411 https://bugs.ruby-lang.org/issues/17724

``` ruby
var = '9801'
def meth = '9801'
# pin operator with local/instance/class/global variable
foo in { bar: ^var }
foo in { bar: ^@var}
foo in { bar: ^@@var}
foo in { bar: ^$var}
# allowed but redundant parens (Style/RedundantParentheses detected; true positive)
foo in { bar: ^(var) }
foo in { bar: ^(@var) }
foo in { bar: ^(@@var) }
foo in { bar: ^($var) }
# pin operator with expression: parens are mandatory (THIS ISSUE)
foo in { bar: ^(var.to_i) }
foo in { bar: ^(meth) }
# syntax error
foo in { bar: ^var.to_i }
```

--------

## Expected behavior

No offences shoud be detected:
``` ruby
# pin.rb
var = '9801'
def meth = '9801'
foo in { bar: ^baz }
foo in { bar: ^(baz.to_i) }
foo in { bar: ^(meth) }
```

## Actual behavior

`Style/RedundantParentheses` detected and autocorrect will remove parens.
```
pin.rb:5:16: C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a method call.
foo in { bar: ^(var.to_i) }
               ^^^^^^^^^^
pin.rb:6:16: C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a method call.
foo in { bar: ^(meth) }
               ^^^^^^
```

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.35.1 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2 x86_64-linux)
```
